### PR TITLE
iox-#885 Configurable introspection update interval

### LINKED ISSUE #885: Make introspection update interval configurable in RouDi
## Brief feature description

At the moment Roudi runs introspections [every second](https://github.com/eclipse-iceoryx/iceoryx/blob/d5f88419b1119b3562de85690ff986c4e6779eb2/iceoryx_posh/include/iceoryx_posh/internal/roudi/introspection/mempool_introspection.hpp#L89). To make it more flexible (e.g. for quicker integration tests) it makes sense to have inrospection update interval configurable.

## Detailed information

* Tests that use [introspection](https://github.com/eclipse-iceoryx/iceoryx/blob/d5f88419b1119b3562de85690ff986c4e6779eb2/iceoryx_posh/test/integrationtests/test_posh_mepoo.cpp#L250) info can benefit, because they don't need to wait for 1 sec each.
* Regular users can also benefit from it, so CLI option may also be added.

### PR DESCRIPTION
**Pull request by r7vme**
## Pre-Review Checklist for the PR Author

1. [ ] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [ ] Tests follow the [best practice for testing][testing]
1. [ ] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [ ] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [ ] Commits messages are according to this [guideline][commit-guidelines]
    - [ ] Commit messages have the issue ID (`iox-#123 commit text`)
    - [ ] Commit messages are signed (`git commit -s`)
    - [ ] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [ ] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [ ] Relevant issues are linked
1. [ ] Add sensible notes for the reviewer
1. [ ] All checks have passed (except `task-list-completed`)
1. [ ] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CHANGELOG.md

## Checklist for the PR Reviewer

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code according to our coding style and naming conventions
- [ ] Unit tests have been written for new behavior
- [ ] Public API changes are documented via doxygen
- [ ] Copyright owner are updated in the changed files
- [ ] PR title describes the changes

## Post-review Checklist for the PR Author

1. [ ] All open points are addressed and tracked via issues

## References

- Closes #885

### CODE REVIEW

File: iceoryx_posh/include/iceoryx_posh/roudi/iceoryx_roudi_components.hpp [hunk 1]
```diff
@@ -29,7 +29,8 @@ namespace roudi
 struct IceOryxRouDiComponents
 {
   public:
-    IceOryxRouDiComponents(const RouDiConfig_t& roudiConfig) noexcept;
+    // TODO: also proxy introspectionUpdateInterval
```
Comment (r7vme) [inline iceoryx_posh/include/iceoryx_posh/roudi/iceoryx_roudi_components.hpp:32]: this still need to be implemented

**Review by r7vme — COMMENTED** [review]
@elBoberido want to have your opinion before polishing this PR. Does it really worth it? There are a lot of smelly places, because we need to proxy this parameter thru many classes.

We can also be pragmatic and drop this feature.

File: iceoryx_posh/source/roudi/iceoryx_roudi_components.cpp [hunk 2]
```diff
@@ -34,7 +36,12 @@ IceOryxRouDiComponents::IceOryxRouDiComponents(const RouDiConfig_t& roudiConfig)
             LogFatal() << "Could not create SharedMemory! Error: " << error;
             errorHandler(Error::kROUDI_COMPONENTS__SHARED_MEMORY_UNAVAILABLE, nullptr, iox::ErrorLevel::FATAL);
         });
-        return &rouDiMemoryManager;
+
+        PortManager::PortManagerConfig portManagerConfig;
+        portManagerConfig.m_roudiMemoryInterface = &rouDiMemoryManager;
+        portManagerConfig.m_introspectionUpdateInterval = units::Duration::fromSeconds(1U);
```
Comment (elfenpiff) [inline iceoryx_posh/source/roudi/iceoryx_roudi_components.cpp:42]: Why not use `INTROSPECTION_DEFAULT_UPDATE_INTERVAL` from the posh types?

**Comment by elBoberido** [comment]
> @elBoberido want to have your opinion before polishing this PR. Does it really worth it? There are a lot of smelly places, because we need to proxy this parameter thru many classes.
> 
> We can also be pragmatic and drop this feature.

At first I thought it would be neat to be able to reduce test times, but since this currently affects only one test we might also just drop it. We would also need to test this feature which would add more code for little gain. So I'm for dropping.

@elfenpiff what do you think?

**Comment by elfenpiff** [comment]
> > @elBoberido want to have your opinion before polishing this PR. Does it really worth it? There are a lot of smelly places, because we need to proxy this parameter thru many classes.
> > We can also be pragmatic and drop this feature.
> 
> At first I thought it would be neat to be able to reduce test times, but since this currently affects only one test we might also just drop it. We would also need to test this feature which would add more code for little gain. So I'm for dropping.
> 
> @elfenpiff what do you think?

I would suggest to keep the `INTROSPECTION_DEFAULT_UPDATE_INTERVAL` in posh types and use this directly in all classes which are affected by it and not provide this as a constructor or method argument.
This should keep the code changes to a minimum but we would not decrease the time spent in tests. But if it is too much effort I also would vote for dropping the feature.

**Comment by r7vme** [comment]
> I would suggest to keep the INTROSPECTION_DEFAULT_UPDATE_INTERVAL in posh types and use this directly in all classes which are affected by it and not provide this as a constructor or method argument.

This should be easy, I'll create a separate PR then.

**Pull request closed without merging**